python_modules/libraries/dagster-aws/dagster_aws/emr/emr_eks_pyspark_step_launcher.py

```diff
@@ -0,0 +1,574 @@
+
+    def _job_event_metadata(self, job_id: str, log_prefix: str) -> Dict[str, MetadataValue]:
+        return {
+            "EMR on EKS Cluster": MetadataValue.url(self._cluster_url()),
+            "\u1217 Job log streams": MetadataValue.url(
+                self._job_log_streams_url(job_id, log_prefix),
+            ),
+            "Job ID": MetadataValue.text(job_id),
+        }
+
+    def _startup_event(self, step_context: StepExecutionContext, job_id: str) -> DagsterEvent:
+        return DagsterEvent(
+            event_type_value=DagsterEventType.ENGINE_EVENT.value,
+            pipeline_name=step_context.dagster_run.pipeline_name,
+            step_key=step_context.step.key,
+            message="Created Job Run",
+            event_specific_data=EngineEventData(
+                metadata_entries=[
+                    MetadataEntry(label=k, value=v)
+                    for k, v in self._job_event_metadata(
+                        job_id, self._sanitize_step_key(step_context.step.key)
+                    ).items()
+                ]
+            ),
+        )
+
+    def _failure_event(
+        self, step_context: StepExecutionContext, exc: EmrEksJobError
+    ) -> DagsterEvent:
+        """Build a Dagster step failure event from a job exception."""
+        application_name = self._application_name(step_context)
+
+        return DagsterEvent(
+            event_type_value=DagsterEventType.STEP_FAILURE.value,
+            pipeline_name=step_context.dagster_run.pipeline_name,
+            step_key=step_context.step.key,
+            message=f"Spark job for '{application_name}' has failed",
+            event_specific_data=StepFailureData(
+                error=None,
+                user_failure_data=UserFailureData(
+                    label="Spark job error",
+                    metadata_entries=[
+                        MetadataEntry(label=k, value=MetadataValue.text(v))
+                        for k, v in dataclasses.asdict(exc).items()
+                    ],
+                ),
+                error_source=ErrorSource.USER_CODE_ERROR,
+            ),
+        )
+
+    def _step_run_ref(
+        self,
+        step_context: StepExecutionContext,
+    ) -> StepRunRef:
+        """Build a StepRunRef that can be used to run the step
+        in a remote process.
+        """
+        step_run_ref = step_context_to_step_run_ref(
+            step_context,
+        )
+
+        if not self.deploy_local_job_package:
+            # Container has the code, but we need to point Dagster to it
+            step_run_ref = EmrEksPySparkResource._set_step_run_ref_working_directory(step_run_ref)
+
+        step_run_ref = EmrEksPySparkResource._remove_parent_run_id(step_run_ref)
+
+        return step_run_ref
+
+    @staticmethod
+    def _set_step_run_ref_working_directory(
+        step_run_ref: StepRunRef,
+        working_directory: str = "/opt/raptor/pipelines/",  # TODO: do not hardcode
```

Comment:
issue: we need to get rid of this. is this workaround still needed?

Reply:
I tried to remove it and things still work